openapi3/schema.go

```diff
@@ -1039,6 +1051,32 @@ func (schema *Schema) visitJSONNumber(settings *schemaValidationSettings, value
 		return schema.expectedType(settings, "number, integer")
 	}
 
+	// formats
+	if schema.Format != "" {
```

Comment:
```suggestion
	if schema.Type == "integer" && schema.Format != "" {
```

Reply:
Yes but used TypeInteger instead of "integer". Also changed  line 1034 to also use TypeInteger